File: 2094_Finding_3_Digit_Even_Numbers.java
class Solution {
    public int[] findEvenNumbers(int[] a) {
        int[] c = new int[10];
        for(int x : a) c[x]++;
        List<Integer> r = new ArrayList<>();
        for(int i=100;i<1000;i+=2){
            int x = i/100, y = (i/10)%10, z = i%10;
            c[x]--; c[y]--; c[z]--;
            if(c[x]>=0 && c[y]>=0 && c[z]>=0) r.add(i);
            c[x]++; c[y]++; c[z]++;
        }
        return r.stream().mapToInt(i->i).toArray();
    }
}

File: 2094_Finding_3_Digit_Even_Numbers.cpp
class Solution {
public:
    vector<int> findEvenNumbers(vector<int>& d) {
        set<int> o;
        int n = d.size();
        for(int i = 0; i < n; ++i)
            for(int j = 0; j < n; ++j)
                for(int k = 0; k < n; ++k)
                    if(i != j && j != k && i != k) {
                        int x = d[i]*100 + d[j]*10 + d[k];
                        if(x > 99 && x % 2 == 0) o.insert(x);
                    }
        return {o.begin(), o.end()};
    }
};
